lib/metasploit/credential/importer/core.rb

```diff
@@ -140,57 +134,105 @@ def import_long_form
             private_object_for_row = private_class.where(data: private_data).first_or_create
           end
         end
+        all_creds_valid = all_creds_valid && public_object && private_object_for_row && (public_object.valid? && private_object_for_row.valid?)
```

Comment:
This should live in a validator. We already have a "header_format_and_csv_wellformednes" validator. What ever logic validation logic you need to validate that the importer is valid should live in there.

You could then call valid? on the instance and get a true or false value returned.